openslides/assignments/views.py

```diff
@@ -242,6 +243,134 @@ def get_serializer_class(self):
             serializer_class = AssignmentShortSerializer
         return serializer_class
 
+    @detail_route(methods=['post', 'delete'])
+    def candidature_self(self, request, pk=None):
+        """
+        View to nominate self as candidate (POST) or withdraw own candidature
+        (DELETE).
+        """
+        if not request.user.has_perm('assignments.can_nominate_self'):
+            self.permission_denied(request)
+        assignment = self.get_object()
+        if assignment.is_elected(request.user):
+            raise ValidationError({'detail': _('You are already elected.')})
+        if request.method == 'POST':
+            message = self.nominate_self(request, assignment)
+        else:
+            # request.method == 'DELETE'
+            message = self.withdraw_self(request, assignment)
+        return Response({'detail': message})
+
+    def nominate_self(self, request, assignment):
+        if assignment.phase == assignment.PHASE_FINISHED:
+            raise ValidationError({'detail': _('You can not candidate to this election because it is finished.')})
+        if assignment.phase == assignment.PHASE_VOTING and not request.user.has_perm('assignments.can_manage'):
+            # To nominate self during voting you have to be a manager.
+            self.permission_denied(request)
```

Comment:
It seems inconsistent that you get a 403 if want to nominate yourself on a assignment with phase == voting, but a 400 on a assignment with phase == finished